INTMDB-594: [CFN-PI] Improve documentations for quickstarts (#480)
mongodb/mongodbatlas-cloudformation-resources: examples/quickstart-mongodb-atlas-analytics-amazon-sagemaker-integration/README.md

# quickstart-mongodb-atlas-analytics-amazon-sagemaker-integration

## Overview

![simple-quickstart-arch](https://user-images.githubusercontent.com/5663078/229119386-0dbc6e30-a060-465e-86dd-f89712b0fc49.png)

> The quickstart template enables you to begin working with your Machine Learning models rapidly, using your MongoDB Atlas Cluster and Amazon SageMaker Endpoint as tools. With this template, you can utilize MongoDB as a data source and Amazon SageMaker for data analysis, streamlining the process of building and deploying machine learning models.

Reviewer: ```suggestion
This Partner Solutions template enables you to begin working with your Machine Learning models rapidly, using your MongoDB Atlas Cluster and Amazon SageMaker Endpoint as tools. With this template, you can utilize MongoDB as a data source and Amazon SageMaker for data analysis, streamlining the process of building and deploying machine learning models.
```

Reviewer: Note AWS Quick Start have been rebranded as AWS Partner Solution Deployments: https://aws.amazon.com/quickstart/?solutions-all.sort-by=item.additionalFields.sortDate&solutions-all.sort-order=desc&awsf.filter-content-type=*all&awsf.filter-tech-category=*all&awsf.filter-industry=*all&solutions-all.q=mongodb&solutions-all.q_operator=AND

> This Partner Solutions template enables you to begin working with your Machine Learning models rapidly, using your MongoDB Atlas Cluster and Amazon SageMaker Endpoint as tools. With this template, you can utilize MongoDB as a data source and Amazon SageMaker for data analysis, streamlining the process of building and deploying machine learning models.

Reviewer 2: ```suggestion
This Partner Solutions template enables you to begin working with your machine learning models using your MongoDB Atlas Cluster and Amazon SageMaker Endpoint as tools. With this template, you can utilize MongoDB as a data source and Amazon SageMaker for data analysis, streamlining the process of building and deploying machine learning models.
```

## MongoDB Atlas CFN Resources used by the templates

> - [MongoDB::Atlas::Trigger](../../cfn-resources/trigger/)

Reviewer: Is this really the only MongoDB Atlas CFN resource used by this template?

Author: yes, the template asks the user to provide ProjectId and collection and DB name. You can search for `MongoDB::Atlas::` in the [template](https://github.com/aws-quickstart/quickstart-mongodb-atlas-analytics-amazon-sagemaker-integration/blob/0ca645280a474cad82443caa3bddb1c6b6c71e39/templates/mongodb-sagemaker-analytics.template.yaml#L98)

> ## Environment Configured by the Quickstart template

Reviewer: ```suggestion
## Environment Configured by the Partner Solutions template
```

Author: Updated

> The Partner Solutions template will generate the following resources:

Reviewer 2: ```suggestion
The Partner Solutions template will generate and configure the following resources:
```

>  - Configure [MongoDB Partner Event Bus](http://mongodb.com/docs/atlas/app-services/triggers/aws-eventbridge/#std-label-aws-eventbridge).
>  - Set up a [database trigger](https://www.mongodb.com/docs/atlas/app-services/triggers/database-triggers/) with your Atlas Cluster.
>  - Create lambda functions to run the machine learning model and send the classification results to your MongoDB Atlas Cluster. 
>     - See [iris_classifier](sagemaker-example/iris_classifier/) for an example of machine learning model to use with this template.
>     -  See [lambda_functions](sagemaker-example/lambda_functions/) for an example of lambda functions to use to read and write data to your MongoDB Atlas cluster.

Reviewer 2: ```suggestion
 - a [MongoDB Partner Event Bus](http://mongodb.com/docs/atlas/app-services/triggers/aws-eventbridge/#std-label-aws-eventbridge)
 - a [database trigger](https://www.mongodb.com/docs/atlas/app-services/triggers/database-triggers/) with your Atlas Cluster
 - lambda functions to run the machine learning model and send the classification results to your MongoDB Atlas Cluster. (See [iris_classifier](sagemaker-example/iris_classifier/) for an example of machine learning model to use with this template. See [lambda_functions](sagemaker-example/lambda_functions/) for an example of lambda functions to use to read and write data to your MongoDB Atlas cluster.)
```

## Permissions required to run the template
See [execution-role.yaml](execution-role.yaml) for the list of permissions needed to run the template. 

> If you want to create an IAM (Identity and Access Management) role to run the quickstart template on a CFN (CloudFormation) stack, you can use the [execution-role.yaml](execution-role.yaml) file. Note that passing an IAM role to CloudFormation when creating a stack is optional. If you don't supply one, the user permissions are assumed. 

Reviewer 2: What are the default permissions? What does "user permissions" refer to?

Author: > What does "user permissions" refer to?

When you log in to the AWS console, you must provide credentials for your AWS user. User permissions are those associated with your AWS user that you used to log in to the AWS console.

> What are the default permissions?

It depends on how you set up your AWS account.

See the [IAM permissions section in the General information guide](https://aws-ia.github.io/content/qs_info.html#_technical_requirements) for more information.

> ## Additional Information

Reviewer: This is a great/helpful section @andreaangiolillo / @poornatejakonathampi !
